### ISSUE TITLE
"Could not create symlink" when using use_unsafe_shared_cache = True

### ISSUE THREAD

**Issue by joprice** [bazelbuild/rules_jvm_external#311]
When activating the `use_unsafe_shared_cache` setting, I see the error below:

```
'@scala_2_11//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_adamjohnson/09e8271e29391b7bad9dc3e7f8bde013/external/rules_jvm_external/coursier.bzl", line 503
		artifact.update(<1 more arguments>)
	File "/private/var/tmp/_bazel_adamjohnson/09e8271e29391b7bad9dc3e7f8bde013/external/rules_jvm_external/coursier.bzl", line 503, in artifact.update
		_relativize_and_symlink_file(<2 more arguments>)
	File "/private/var/tmp/_bazel_adamjohnson/09e8271e29391b7bad9dc3e7f8bde013/external/rules_jvm_external/coursier.bzl", line 69, in _relativize_and_symlink_file
		repository_ctx.symlink(<2 more arguments>)
java.io.IOException: Could not create symlink from /Users/adamjohnson/Library/Caches/Coursier/v1/http/localhost%3A8080/iheart/amp-bazel-external/com/fasterxml/jackson/core/jackson-databind/2.7.8/jackson-databind-2.7.8.jar to /private/var/tmp/_bazel_adamjohnson/09e8271e29391b7bad9dc3e7f8bde013/external/scala_2_11/v1/http/localhost%3A8080/iheart/amp-bazel-external/com/fasterxml/jackson/core/jackson-databind/2.7.8/jackson-databind-2.7.8.jar: /private/var/tmp/_bazel_adamjohnson/09e8271e29391b7bad9dc3e7f8bde013/external/scala_2_11/v1/http/localhost%3A8080/iheart/amp-bazel-external/com/fasterxml/jackson/core/jackson-databind/2.7.8/jackson-databind-2.7.8.jar (File exists)
```

I have tried deleting the cache and re-running and get the same error.

**Comment by joprice** [comment]
This seems to be caused by the dependency tree having dupes on https://github.com/bazelbuild/rules_jvm_external/blob/0e365338b26ff9879ab67683730f50a184b3f3be/coursier.bzl#L493.

**Comment by jin** [comment]
Do you have a `maven_install` I can reproduce this on?

**Comment by joprice** [comment]
Not at hand, but I can try to minimize one that doesn't contain internal repos.

**Comment by gergelyfabian** [comment]
I received the same error and it made me create PR #325 (that solves imho also another issue related to Bazel cache safety, but was also this issue is not present with that solution).

**Comment by jin** [comment]
@gergelyfabian can you provide a repro please? Looks like `jackson-databind` triggers this, but it'll help for me to repro the actual bug.

**Comment by joprice** [comment]
Here's a repro https://github.com/joprice/bazel_experiments/tree/issue-311. Run `bazel run @unpinned_scala_2_11//:pin` to produce the error. The two example dependencies that trigger it are "com.typesafe.play:play_2.11:2.5.19" and "org.scalatestplus.play:scalatestplus-play_2.11:2.0.1".

**Comment by jin** [comment]
Thanks for the repro, I see the issue. Fix in #327
